enhancements/user-defined-default-ingress-controller.md

```diff
@@ -0,0 +1,262 @@
+---
+title: user-defined-default-ingress-controller
+authors:
+  - "@ironcladlou"
+reviewers:
+  - "@ironcladlou"
+  - "@Miciah"
+  - "@danehans"
+approvers:
+  - "@ironcladlou"
+  - "@Miciah"
+  - "@danehans"
+creation-date: 2019-09-08
+last-updated: 2019-09-09
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# User-defined Default Ingress Controller
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal is to enable users to define the default
+IngressController<sup>[1](https://docs.openshift.com/container-platform/4.1/networking/configuring-ingress-cluster-traffic/configuring-ingress-cluster-traffic-ingress-controller.html#configuring-ingress-cluster-traffic-ingress-controller),[2](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go)</sup>
+during OpenShift installation.
+
+Changes to the OpenShift [Installer](https://github.com/openshift/installer) and
+[Ingress Operator](https://github.com/openshift/cluster-ingress-operator) are
+proposed which will provide the basis for a supportable customization procedure.
+The procedure is proposed for incorporation into the OpenShift
+[documentation](https://github.com/openshift/openshift-docs).
+
+## Motivation
+
+OpenShift's default IngressController is automatically created based on the
+cluster platform and cluster-global configuration. The default is reliable for
+installation, but typically requires reconfiguration soon after installation.
+
+However, there are some updates to an existing IngressController which are
```

Comment:
k8s-networking changes are disruptive and destructive i.e you can't really change the type from sdn to ovn today after installation, changing the service network is difficult.

How are the default ingress changes disruptive and destructive, such that they need to be modified during installation??

Reply:
>How are the default ingress changes disruptive and destructive, such that they need to be modified during installation??

This is covered in each user story below. https://github.com/openshift/enhancements/pull/31/files#diff-40bb87d9f5690ccc8c6d75986d956f45R87

Reply:
>k8s-networking changes are disruptive and destructive i.e you can't really change the type from sdn to ovn today after installation, changing the service network is difficult.

I agree. However, changing ingress is somewhat less disruptive and difficult than the service network, and can also involve cost considerations for cloud resources which are unnecessarily provisioned. So, it seems worth considering.